test/controllers/authorizeService.js

```diff
@@ -0,0 +1,424 @@
+var proxyquire = require('proxyquire').noCallThru();
+
+describe('Accounting Service', function () {
+
+	var DEFAULT_URL = 'http://example/path';
+	var DEFAULT_APIKEY = 'apiKey';
+
+	var getAuthorizeServiceController = function (accServiceSchema, config) {
+		return proxyquire('../../controllers/authorizeService', {
+			'../db/schemas/accountingService': accServiceSchema,
+			'../config': config
+		}).authorizeService;
+	};
+
+	describe('Get api-key', function () {
+
+		it('should return 401 when the requester is not the WStore', function (done) {
+			var WStoreHost = 'localhost';
+			var WStorePort = '8080';
+
+			var config = {
+				appHost: WStoreHost,
+				endpoints: {
+					charging: {
+						port: WStorePort
+					}
+				}
+			};
+
+			var accSerivceController = getAuthorizeServiceController({}, config);
+
+			var req = {
+				ip: {
+					replace: function (expr) { return '130.25.13.12'}
+				},
+				connection: {
+					remotePort: 9065
+				}
+			};
+			var res = {
+				status: function (code) {
+					return this;
+				},
+				json: function (err) {}
+			};
+			spyOn(res, 'status').and.callThrough();
+			spyOn(res, 'json').and.callThrough();
+
+			accSerivceController.getApiKey(req, res);
+
+			setTimeout(function () {
+
+				expect(res.status).toHaveBeenCalledWith(401);
+				expect(res.json).toHaveBeenCalledWith({ error: 'Invalid remote client' });
+
+				done();
+
+			}, 100);
+		});
+
+		it('should return 400 when the "url" is not defined', function (done) {
+			var WStoreHost = 'localhost';
```

Comment:
The code of this tests is very similar to the previous one. DRY (Do not Repeat Yourself!). Use functions! ;)